Add browser async breaking change (#3630)
grafana/k6: release notes/v0.50.0.md

## Breaking changes

### Browser APIs to Async

> In future release we are going to be moving most of the browser APIs to asynchronous (promisfying them). We expect that this will affect majority of our users, which is why we are posting this upfront before we make the change itself. Here are the reasons for making this large breaking change:

Reviewer: How about?

```suggestion
In future releases, we are going to be moving most of the browser APIs to asynchronous (promisifying them). We expect this will affect most of our users, so we are posting this upfront before making the change. Here are the reasons for making this large breaking change:
```

> 1. Most of the browser APIs use some form of long running IO operation (networking) to perform the requested action on the web browser against the website under test. We need to avoid blocking the javascript's runtime event loop for such operations.

Reviewer: ```suggestion
1. Most browser APIs use some form of long-running IO operation (networking) to perform the requested action on the web browser against the website under test. We need to avoid blocking Javascript's runtime event loop for such operations.
```

> 2. We're going to be adding more asynchronous event based APIs (such as [page.on](https://github.com/grafana/xk6-browser/issues/1227)) that would be blocked with our current synchronous APIs.

Reviewer: ```suggestion
2. We're going to add more asynchronous event-based APIs (such as [page.on](https://github.com/grafana/xk6-browser/issues/1227)) that our current synchronous APIs would block.
```

> 3. To align with how javascript developers expect to work with javascript APIs.

Reviewer: ```suggestion
3. To align with how developers expect to work with JavaScript APIs.
```
